src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,207 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+
+        if (size == 0) {
+            addFirstElementToList(value);
+            return;
+        }
+
+        if (index == 0) {
+            addFirst(value);
+            return;
+        }
+        if (index == size) {
+            addLast(value);
+            return;
+        }
+
+        insertNode(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
+        checkIndexAdvanced(index);
+
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if (i == index) {
+                return node.value;
+            }
+            node = node.next;
+        }
```

Comment:
It will be better if you will check if index is lower than size / 2 and then u can decide iterate from tail or iterate from head. It will improve your linkedList performance.

https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#dont-iterate-your-nodes-from-the-head-to-tail-or-vice-versa